File: FrontEnd/src/app/features/billing/invoicesViewer/classes/view-models/viewer/invoiceViewer-vm.ts
import { Guid } from 'guid-typescript'
// Custom
import { AadeVM } from './aade-vm'
import { DocumentTypeVM } from 'src/app/features/billing/invoices/classes/view-models/list/documentType-vm'
import { InvoiceViewerPartyTypeVM } from './invoiceViewer-partyType-vm'
import { InvoiceViewerPortVM } from '../../dtos/form/invoiceViewer-port-vm'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface InvoiceViewerVM {

    // PK
    invoiceId: Guid
    // Fields
    date: string
    invoiceNo: number
    // Object fields
    customer: InvoiceViewerPartyTypeVM
    destination: SimpleEntity
    documentType: DocumentTypeVM
    paymentMethod: SimpleEntity
    ship: SimpleEntity
    aade: AadeVM
    issuer: InvoiceViewerPartyTypeVM
    invoicesPorts: InvoiceViewerPortVM[]
    // Fields
    remarks: string
    // Persons
    adults: number
    kids: number
    free: number
    totalPax: number
    // Amounts
    netAmount: number
    vatPercent: number
    vatAmount: number
    grossAmount: number

}

File: API/Features/Billing/Invoices/Dtos/Viewer/InvoiceViewerVM.cs
using System;
using System.Collections.Generic;

namespace API.Features.Billing.Invoices {

    public class InvoiceViewerVM {

        public Guid InvoiceId { get; set; }
        public string Date { get; set; }
        public int InvoiceNo { get; set; }
        public decimal NetAmount { get; set; }
        public decimal VatPercent { get; set; }
        public decimal VatAmount { get; set; }
        public decimal GrossAmount { get; set; }
        public string Remarks { get; set; }
        public string Ship { get; set; }
        public InvoiceViewerAadeVM Aade { get; set; }
        public List<InvoiceViewerPortVM> InvoicesPorts { get; set; }
        public InvoiceViewerPartyVM Customer { get; set; }
        public string Destination { get; set; }
        public InvoiceViewerDocumentTypeVM DocumentType { get; set; }
        public string PaymentMethod { get; set; }
        public InvoiceViewerPartyVM Issuer { get; set; }

    }

}